observability/test/test_utils.py

```diff
@@ -0,0 +1,177 @@
+
+    def __init__(self, config: Dict[str, Any]) -> None:
+        self.config = config
+
+    def setdefault(self, key: str, value: Any) -> None:
+        self.config.setdefault(key, value)
+
+    def set(self, key: str, value: Any) -> None:
+        self.config[key] = value
+
+    def toJson(self) -> str:
+        return json.dumps(self.config)
+
+class TestRunMetadata:
+    script_start_time: float
+    script_start_seconds: int
+    nanos: int
+
+    def __init__(self) -> None:
+        self.script_start_time = time.time()
+        self.script_start_seconds = int(self.script_start_time)
+        self.nanos = int((self.script_start_time - self.script_start_seconds) * 10**9)
+
+class LoggerSide(str, Enum):
+    SERVER = 1
+    CLIENT = 2
+
+    def __str__(self) -> str:
+        return self.name
+
+class ExpectCount(Enum):
+    ZERO = 1
+    AT_LEAST_ONE = 2
+
+class ConfigLocation(Enum):
+    FILE = 1
+    ENV_VAR = 2
+
+class ServerStartArgs:
+    port: str
+
+    def __init__(self,
+                 port: str) -> None:
+        self.port = port
+
+class ClientActionArgs:
+    server_host: str
+    server_port: str
+    exporter_interval: int
+    action: str
+
+    def __init__(self,
+                 server_host: str,
+                 server_port: str,
+                 exporter_interval: int,
+                 action: str) -> None:
+        self.server_host = server_host
+        self.server_port = server_port
+        self.exporter_interval = exporter_interval
+        self.action = action
+
+class LanguageConstants:
+    repo_name: str
+    server_start_cmd: str
+    server_start_args: str
+    client_action_cmd: str
+    client_action_args: str
+
+    def __init__(self,
+                 repo_name: str,
+                 server_start_cmd: str,
+                 server_start_args: str,
+                 client_action_cmd: str,
+                 client_action_args: str) -> None:
+        self.repo_name = repo_name
+        self.server_start_cmd = server_start_cmd
+        self.server_start_args = server_start_args
+        self.client_action_cmd = client_action_cmd
+        self.client_action_args = client_action_args
+
+    def get_server_start_args_replaced(self, args: ServerStartArgs) -> str:
+        return self.server_start_args.format(
+            port = args.port)
+
+    def get_client_action_args_replaced(self, args: ClientActionArgs) -> str:
+        return self.client_action_args.format(
+            server_host = args.server_host,
+            server_port = args.server_port,
+            exporter_interval = args.exporter_interval,
+            action = args.action)
+
+class TestUtil:
+    @staticmethod
+    def get_sponge_log_dir(job_mode: JobMode, job_name: str) -> str:
+        artifacts_dir = os.environ.get('KOKORO_ARTIFACTS_DIR')
+        if not artifacts_dir:
+            raise ValueError('Env var KOKORO_ARTIFACTS_DIR is not defined')
+        sponge_log_dir = os.path.join(artifacts_dir, str(job_mode), job_name)
```

Comment:
Nit: Why the `str(job_mode)`? Isn't it already a `str`?

Reply:
Fixed.